### DISCUSSION TITLE
Indoor Temp population errors - Not strictly a bug

### DISCUSSION THREAD

**Discussion by lloydbayley** [Blackymas/NSPanel_HA_Blueprint#1837] [Q&A] [answered]
G'day all,

Since the last update, something peculiar is being logged in HA.

```
Logger: homeassistant.components.automation.nspanel_configuration
Source: helpers/script.py:1805
Integration: Automation ([documentation](https://www.home-assistant.io/integrations/automation), [issues](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+automation%22))
First occurred: 10:49:46 PM (4 occurrences)
Last logged: 10:49:46 PM

    Populate - nspanel_eu_blk: Main choices: Indoor temp - Entity: If at step 1: If at step 2: Error executing script. Invalid data for call_service at pos 6: expected a list for dictionary value @ data['foreground']
    Populate - nspanel_eu_blk: Main choices: Indoor temp - Entity: If at step 1: Error executing script. Invalid data for if at pos 2: expected a list for dictionary value @ data['foreground']
    Populate - nspanel_eu_blk: Main choices: Indoor temp - Entity: Error executing script. Invalid data for if at pos 1: expected a list for dictionary value @ data['foreground']
    Populate - nspanel_eu_blk: Error executing script. Invalid data for choose at pos 2: expected a list for dictionary value @ data['foreground']
```

It seems that it does indeed affect the indoor temp display because it's not correct. Everything else about the panel seems fine.

I've been through the automation and can't spot anything obvious. It's definitely the automation because if I create a new one, the problem is not appearing. However, being a programmer myself, I'd like to understand the nature of the problem and not just create a new automation and never know.

I've been through some of the code but everything I look at and then check my automation, seems fine.
I could understand if I was playing with the automation but I haven't touched it in a while.
Has anyone else suddenly come across this kind of error?

**Comment by edwardtfn** [comment, marked as answer]
We have changed the color format a long time ago (I believe o v3.3) and looks like you have one of your inputs with the old selection type, which was a number and not the native color picker. When creating a new automation you probably selected the color again and it was accordingly to the new format.
If you still having the older automation, you shoukd be able to find the color selector missing a selection.

**Reply by lloydbayley** [reply]
Thanks Edward, I was playing with those to see if the format somehow became corrupt but I couldn't see anything. Will have another careful go at it. It's good to confirm that I was on the right track.
I'm sure I'll find it at some stage soon.

**Comment by lloydbayley** [comment]
Yep, there was a sneaky colour number tucked away in there. Took a while to find it. Kept missing it. Thanks for the pointer in the right direction! :)
